tests/mtest/src/main/scala/tests/Assertions.scala

```diff
@@ -57,6 +58,24 @@ trait Assertions extends munit.Assertions {
     }
   }
 
+  def assertReferencesInOrder(locations: List[lsp4j.Location]): Unit = {
+
+    val grouped = locations.groupBy(_.getUri)
+    def sortByLocationPosition(
+        l1: lsp4j.Location,
+        l2: lsp4j.Location
+    ): Boolean = {
+      l1.getRange.getStart.getLine < l2.getRange.getStart.getLine
+    }
+
+    grouped.foreach {
+      case (_, group) =>
+        assertNoDiff(
```

Comment:
So I'm a bit stuck with this. I feel like this isn't doing anything since I'm just using the same logic to sort in `ReferenceProvider` as we are to test it. I know this isn't right. However, again, the way the tests are, it doesn't make it easy to be explicit about what exactly is going to be returned. Do you think it makes sense to change the input of `check` to include another parameter that is just a string representation of the expected references to ensure that they are in order?